internal/supervise/main.go

```diff
@@ -156,41 +156,55 @@ func Main() {
 
 func pipeToSyslog(ctx context.Context, conn net.Conn, componentID string, name string, procID string, r io.Reader) {
 	b := bufio.NewReaderSize(r, api.MaxMessageSize)
-	for {
+	readLine := func() (string, error) {
+		// Usage of ReadLine in preference to ReadString is intentional, since
+		// ReadString will perform unbounded buffering.
+		// See discussion here: https://github.com/deref/exo/pull/322
 		message, isPrefix, err := b.ReadLine()
-		if errors.Is(err, io.EOF) || errors.Is(err, os.ErrClosed) {
-			return
-		}
-		if err != nil {
-			fatalf("reading %s: %v", name, err)
-		}
-		// TODO: Do something better with lines that are too long.
 		for isPrefix {
 			// Skip remainder of line.
 			message = append([]byte{}, message...)
```

Comment:
blame suggests I wrote this line, but I have no idea what it is trying to do... Why is it copying the message, but only if `isPrefix` seems like this line could just be deleted?